[ROCm] Skipping subtests that check support for float64 type in the NN ops

deven-amd opened the pull request:
ROCm platform currently does not support the float64/double type in the NN ops

This commit skips subtests (within python unit-tests) that test this functionality. The "skip" is guarded by the call to "is_built_with_rocm()", and hence these unit-tests will not be affected in any way when running with TF which was not built with ROCm support (i.e. `--config=rocm`)

---------------------------------

@tatianashp @whchung @chsigg

whchung commented on tensorflow/python/ops/nn_test.py:
```diff
@@ -1445,6 +1470,7 @@ def test2DTensor(self):
     self.assertAllEqual(self.evaluate(y1), self.evaluate(y2))
 
   def test3D(self):
+
```
nit: please help remove these empty lines.

deven-amd replied:
done

whchung reviewed (changes requested):
@deven-amd minor cosmetic changes are required.

deven-amd committed:
[ROCm] Skipping subtests that check support for float64 type in the NN ops

ROCm platform currently does not support the float64/double type in the NN ops

This commit skips subtests (within python unit-tests) that test this functionality. The "skip" is guarded by the call to "is_built_with_rocm()", and hence these unit-tests will not be affected in any way when running with TF which was not built with ROCm support (i.e. `--config=rocm`)

deven-amd commented:
I looked at the CI failure logs, and they do not seem related to the changes in this PR.

The one failure in `Linux GPU` is in a test that was modified by this PR. However the changes in this PR should be a no-op for the CUDA / `Linux GPU` run, and hence it does not seem likely that the change in this PR is the cause for the failure.

rthadur commented:
@deven-amd here is the internal error : 
`InternalError: 2 root error(s) found.
  (0) Internal: cuDNN Backward Data function launch failure : input shape([1,5,8,7,2]) filter shape([1,2,3,2,3])
	 [[node gradients_5/conv_5_grad/Conv3DBackpropInputV2 (defined at /third_party/py/absl/third_party/unittest3_backport/case.py:162) ]]
  (1) Internal: cuDNN Backward Data function launch failure : input shape([1,5,8,7,2]) filter shape([1,2,3,2,3])
	 [[node gradients_5/conv_5_grad/Conv3DBackpropInputV2 (defined at /third_party/py/absl/third_party/unittest3_backport/case.py:162) ]]
	 [[gradients_5/conv_5_grad/Conv3DBackpropInputV2/_19]]`

deven-amd commented:
@rthadur 

I saw the error in the invocation log, but cannot see how the change in this PR can cause it. 

For the CUDA run, the only thing that will change with this PR is the ordering the dtypes in the list, it will go from `[dtypes.float64, dtypes.float32, dtypes.float16]` to `[dtypes.float32, dtypes.float16, dtypes.float64]`. The two are functionally equivalent and should not cause any failure (I would think)

Let me push out a change that keeps the above mentioned ordering intact, and see it that fixes the error!

deven

deven-amd commented:
@whchung , please re-approve this to kick-off the CI runs. 
I need to figure out whether or not my last change fixes the failure in the `LInux GPU` CI run.

Thanks
deven

deven-amd commented:
@rthadur , the CI errors are gone :)

rthadur commented:
@deven-amd thank you , @chsigg can you please review this again.
